### ISSUE TITLE
Consider moving to a database for questions

### ISSUE THREAD

**Issue by austintackaberry** [austintackaberry/ydkjs-exercises#137]
So now we have unique ids for each questions which is nice, but there are still some issues. Let's say that User A answered a question with an id of `11111111111111111111` correctly.

Now what if we delete that question, but then some time later we add another question. We notice that `11111111111111111111` is available for use because no other questions have that id, so we assign that id to our new question.

Now when that user goes back to our site, our app will think that this user has already answered that new question correctly, because it finds a question with an id of `11111111111111111111` is already answered.

Basically what this means is that we need to keep track of the ids of all deleted questions and ensure that we always make new, unique ids for new questions. This is much easier to do in a database. Thoughts?

**Comment by austintackaberry** [comment]
I don't think that would necessarily solve the problem. We would still have to keep all deleted questions in `/data` or keep a file that has all `questionIds` that were associated with deleted questions which would kind of be like maintaining a database.

**Comment by nikrb** [comment]
oh I'm not with you then ... are you considering adding a backend?

**Comment by austintackaberry** [comment]
A backend might be nice for handling auth, but I'm not sure how to have a database with open source software.

**Comment by austintackaberry** [comment]
Closing this issue since we decided to not worry about this right now
